src/autolabel/transforms/pdf.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Dict, Any
+
+from pdf2image import convert_from_path
+import pytesseract
+
+from autolabel.transforms import BaseTransform
+
+
+class PDFTransform(BaseTransform):
+    def __init__(
+        self,
+        output_columns: List[str],
+        file_path_column: str,
+        page_header: str = "Page {page_num}: ",
+        page_sep: str = "\n\n",
+    ) -> None:
+        """The output columns for this class should be in the order: [content_column, num_pages_column]"""
+        super().__init__(output_columns)
+        self.file_path_column = file_path_column
+        self.page_header = page_header
+        self.page_sep = page_sep
+
+    @staticmethod
+    def name() -> str:
+        return "pdf"
+
+    @staticmethod
+    def extract_text(path: str) -> List[str]:
+        """This function extracts text from a PDF file using the pdf2image and pytesseract libraries.
+
+        Args:
+            path (str): The path to the PDF file.
+
+        Returns:
+            List[str]: A list of strings, one for each page of the PDF file.
+        """
+        pages = convert_from_path(path)
+        texts = []
+        for page in pages:
+            text = pytesseract.image_to_string(page)
+            texts.append(text)
+        return texts
+
+    def transform(self, row: Dict[str, any]) -> Dict[str, any]:
+        """This function transforms a PDF file into a string of text. It uses the PyPDFLoader to load and split the PDF into pages.
+        Each page is then converted into text and appended to the output string.
+
+        Args:
+            row (Dict[str, any]): The row of data to be transformed.
+
+        Returns:
+            Dict[str, any]: The transformed row of data.
+        """
+        try:
+            from langchain.document_loaders import PyPDFLoader
+        except ImportError:
+            raise ImportError(
+                "pypdf is required to use the pdf transform. Please install pypdf with the following command: pip install pypdf"
+            )
+        pages = self.extract_text(row[self.file_path_column])
```

Comment:
hmm, I'm trying to understand how the langchain pypdf loader and tesseract OCR. the ideal behavior imo is to make native pdf document loading the default, and go the OCR route only if a "ocr_enabled" flag is turned on

Reply:
+1

Also, lets try to import on transform init instead of the .transform method so that once a transform object is initialized we know that we have the necessary things to run the transform